apiserver/common/networkingcommon/linklayer.go

```diff
@@ -6,13 +6,17 @@ package networkingcommon
 import (
 	"github.com/juju/collections/set"
 	"github.com/juju/errors"
+	"github.com/juju/juju/state"
 	"gopkg.in/mgo.v2/txn"
 
 	"github.com/juju/juju/core/network"
 )
 
 // LinkLayerDevice describes a single layer-2 network device.
 type LinkLayerDevice interface {
+	// DocID returns the globally unique identifier for the device.
+	DocID() string
```

Comment:
Can we call this something else, e.g. `ID`? Doc alludes to the underlying implementation detail of the `state` package

Reply:
I would actually prefer to leave it for that reason. It highlights the inadequacy of identifiers (as in implementation agnostic IDs ala machine IDs) for these devices.

It is used (not in this patch) to match `ParentID`s also from this indirection, which must be a generated due to having inconsistent format.

I will put a comment to the effect that we should not _have_ to expose it and why.

Reply:
Shouldn't this just be yet another opaque ID as far as the facade is concerned? I agree with adding a comment about the nature of the ID but I still believe that an interface (especially in an unrelated package to state) should not leak information about the underlying implementation as this makes it harder to change things in the future. 

For example, what if juju 3 switches to a different non-document DB that uses UUIDs instead? We would then need to go back and refactor the interface so it makes sense, update the call-sites and regenerate the mocks; I think that would require considerable more effort than using a proper naming convention from the start (also makes it consistent with the Machine interface below).

However, this is more of a pedantic comment and doesn't really have any effect to the changes from this PR so if you still prefer to leave it as-is, I don't mind approving the PR.

Reply:
OK; see the last commit. The state type has an `ID` method that omits the model UUID, and `ParentID` has been made congruent with `ID`.